jujugui/static/gui/src/app/components/deployment/choose-cloud/choose-cloud.js

```diff
@@ -33,7 +33,20 @@ YUI.add('deployment-choose-cloud', function() {
     }],
 
     propTypes: {
-      changeState: React.PropTypes.func.isRequired
+      changeState: React.PropTypes.func.isRequired,
+      jem: React.PropTypes.object.isRequired
+    },
+
+    getInitialState: function() {
+      return {
+        credentials: []
+      };
+    },
+
+    componentWillMount: function() {
+      this.props.jem.listTemplates((error, data) => {
+        this.setState({credentials: data});
```

Comment:
Shouldn't this do something reasonable in the error state? console.error and return, or something?